apps/web/src/app/[locale]/challenge/[id]/solutions/_components/_actions.ts

```diff
@@ -26,6 +26,21 @@ export async function postSolution({ challengeId, description, title, userId }:
   revalidateTag(`challenge-${challengeId}-submissions`);
 }
 
+export async function deleteSolution(id: number) {
+  const session = await getServerAuthSession();
+  const solution = await prisma.sharedSolution.findUnique({
+    where: { id }
+  })
+  if (!isAuthor(session, solution?.userId)) {
+    throw new Error('Only author can delete their solution.')
+  }
+
+  await prisma.sharedSolution.delete({
+    where: { id }
+  })
+  revalidateTag(`challenge-${id}-submissions`);
```

Comment:
AFAIK you should only have to do this to get the data to update after deleting.

https://nextjs.org/docs/app/api-reference/functions/revalidateTag

Reply:
Ya, It doesn't have any effect here either, I've to try something to refetch/revalidate data after deletion.